src/Microsoft.Extensions.DependencyInjection.Specification.Tests/DependencyInjectionSpecificationTests.cs

```diff
@@ -158,6 +158,33 @@ public void MultipleServiceCanBeIEnumerableResolved()
         }
 
         [Fact]
+        public void RegistrationOrderIsPreservedWhenServicesAreIEnumerableResolved()
```

Comment:
This is one of the expectations that I point out is already tested in https://github.com/aspnet/DependencyInjection/issues/379#issuecomment-225053268

>- If a single instance is requested using GetRequiredService<T>(), the container is expected to return the last registration for T, in case there are multiple registrations for T.
>  - [LastServiceReplacesPreviousServices](https://github.com/aspnet/DependencyInjection/blob/ed7e194b02e42e1bbc608c3222913d899f588e83/src/Microsoft.Extensions.DependencyInjection.Specification.Tests/DependencyInjectionSpecificationTests.cs#L254)